website/README.md

```diff
@@ -68,6 +80,12 @@ Before sending the PR for review, please run:
 
     $ git checkout -- content
 
+## Website generation
+
+After a PR is merged, a background Jenkins job will run and update the website's
+[generated-content
+directory](https://github.com/apache/beam/tree/asf-site/website/generated-content).
```

Comment:
Mention in text that this is on the `asf-site` branch?

"After a PR is merged, a background Jenkins job will run to push generated HTML to the asf-site branch"